modules/assim.sequential/R/sda_plotting.R

```diff
@@ -325,85 +326,97 @@ post.analysis.ggplot <- function(settings, t, obs.times, obs.mean, obs.cov, obs,
-               Type="Data",
-               Date=one.day.data$Date%>%as.POSIXct())
-      
-      
-    })%>%
-    #filter(Variables %in% var.names)%>%
+  
+  tryCatch({
+      ready.OBS<- names(obs.mean)%>%
+        purrr::map(~c(obs.mean[.x],obs.cov[.x],.x)%>%
+                     setNames(c('means','covs','Date')))%>%
+        setNames(names(obs.mean))%>%
+        purrr::map_df(function(one.day.data){
+          #CI
+          
+          purrr::map2_df(sqrt(diag(one.day.data$covs)), one.day.data$means,
+                         function(sd, mean){
+                           data.frame(mean-(sd*1.96), mean+(sd*1.96))
+                           
+                         })%>%
+            mutate(Variables=names(one.day.data$means))%>%
+            `colnames<-`(c('2.5%','97.5%','Variables'))%>%
+            mutate(means=one.day.data$means%>%unlist,
+                   Type="Data",
+                   Date=one.day.data$Date%>%as.POSIXct(tz="EST"))
+          
+          
+        })
+    },
+    error = function(e) {
+      ready.OBS<-NULL
+    }
+  )
+
+  ready.to.plot <- ready.OBS %>%
     bind_rows(ready.FA)
   
-  ready.to.plot$Variables%>%unique()%>%
-    purrr::map(function(vari){
-      
-      varin<-vari
-      unit<-""
-      if (substr(vari,1,8)=="AGB.pft.") varin <- "AGB.pft"
-      #finding the unit
+  #Adding the units to the variables
+  ready.to.plot$Variable %>% unique() %>% 
+    walk(function(varin){
+      #find the unit
       unitp <- which(lapply(settings$state.data.assimilation$state.variable, "[", 'variable.name') %>% unlist %in% varin)
-      if (length(unitp)>0) unit <- settings$state.data.assimilation$state.variable[[unitp]]$unit
-      #plotting
-      ready.to.plot%>%
-        filter(Variables==vari)%>%
+      if (length(unitp)>0) {
+        unit <- settings$state.data.assimilation$state.variable[[unitp]]$unit
+        
+        #replace it in the dataframe
+        ready.to.plot$Variable[ready.to.plot$Variable==varin] <<- paste(varin,"(",unit,")")
+      }
+      
+    })
+  
+  
+
+      p<-ready.to.plot%>%
         ggplot(aes(x=Date))+
         geom_ribbon(aes(ymin=`2.5%`,ymax=`97.5%`,fill=Type),color="black")+
         geom_line(aes(y=means, color=Type),lwd=1.02,linetype=2)+
         geom_point(aes(y=means, color=Type),size=3,alpha=0.75)+
         scale_fill_manual(values = c(alphapink,alphagreen,alphablue),name="")+
         scale_color_manual(values = c(alphapink,alphagreen,alphablue),name="")+
         theme_bw(base_size = 17)+
-        labs(y=paste(vari,'(',unit,')'))+
+        facet_wrap(~Variables, scales = "free", ncol=2)+
         theme(legend.position = "top",
               strip.background = element_blank())->p
       if (!is.null(plot.title)) p <- p + labs(title=plot.title)
-      p
-    })->all.plots
+
+
   
   
-  pdf("SDA/SDA.pdf",width = 10,height = 8)
-  all.plots %>% purrr::map(~print(.x))
+  pdf("SDA/SDA.pdf", width = 14, height = 10, onefile = TRUE)
+  print(p)
```

Comment:
do you want print here or like ggsave?

Reply:
I could do both. But here I moved from each state variable having their separate page to all being next other in one large page for comparison. Right after this line I also write down the time series and the plot objects for users to customize it if they want.